### PR TITLE
e2e: marriage

### PR DESCRIPTION
**Pull request by Eezi**

### CODE REVIEW

**Comment by Eezi** [comment]
- [x] Id selectors should be changed to text or label.

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        page.getByText('Please select the type of event', { exact: true })
+      ).toBeVisible()
+    })
+
+    await test.step('1.1.3 Select the "Marriage" event and click "Continue" button', async () => {
+      await page.getByText('Marriage', { exact: true }).click()
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(
+        page.locator('#form_section_id_who-is-applying-view-group')
+      ).toBeVisible()
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts:39]:
```suggestion
        page.getByText("Informant's details", { exact: true })
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      await expect(
+        page.getByText('Please select the type of event', { exact: true })
+      ).toBeVisible()
+    })
+
+    await test.step('1.1.3 Select the "Marriage" event and click "Continue" button', async () => {
+      await page.getByText('Marriage', { exact: true }).click()
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(
+        page.locator('#form_section_id_who-is-applying-view-group')
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
        page.getByText("Informant's details", { exact: true })
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      await expect(page.locator('label', { hasText: 'Email' })).toBeVisible()
+    })
+    // 1.3.2 Is missing because "Informant details" page does not give an error if
+    // user click continue and nothing is selected.
+    await test.step('1.3.3 Select any option in Informant type > Click Continue', async () => {
+      await page.locator('#informantType').click()
+      await page.getByText('Groom', { exact: true }).click()
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(
+        page.locator('#form_section_id_groom-view-group')
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
        page.getByText("Groom's details", { exact: true })
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      ).toBeVisible()
+    })
+    await test.step('1.4. Validate Groom Details page', async () => {
+      await validateSectionButtons(page)
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(
+        page.locator('#form_section_id_bride-view-group')
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
        page.getByText("Bride's details", { exact: true })
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+      ).toBeVisible()
+    })
+    await test.step('1.5. Validate Bridge Details page', async () => {
+      await validateSectionButtons(page)
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(
+        page.locator('#form_section_id_marriage-event-details')
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
        page.getByText('Marriage details', { exact: true })
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+      ).toBeVisible()
+    })
+    await test.step('1.6. Validate Marriage Details page', async () => {
+      await validateSectionButtons(page)
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(
+        page.locator('#form_section_id_witness-view-group')
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
        page.getByText('Witness 1 details', { exact: true })
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+      ).toBeVisible()
+    })
+    await test.step('1.7. Validate witness 1 Details page', async () => {
+      await validateSectionButtons(page)
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(
+        page.locator('#form_section_id_witness-view-group')
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
        page.getByText('Witness 2 details', { exact: true })
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+      ).toBeVisible()
+    })
+    await test.step('1.8. Validate witness 2 Details page', async () => {
+      await validateSectionButtons(page)
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(
+        page.locator('#form_section_id_documents-view-group')
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
        page.getByText('Upload supporting documents', { exact: true })
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+      ).toBeVisible()
+    })
+    await test.step('1.9. Validate Supporting document page', async () => {
+      await validateSectionButtons(page)
+      await page.getByText('Continue', { exact: true }).click()
+      await expect(page.locator('#review_header')).toBeVisible()
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
      page.getByText('Register event', { exact: true })
```

**Pull request merged**